src/utils/hooks/transaction/use-transaction.ts

```diff
@@ -1,102 +1,116 @@
+) => MutationFunction<TransactionResult, TransactionActions> = (feeAmount, pools) => async (params) => {
   const expectedStatus = params.customStatus || getStatus(params.type);
   const baseExtrinsic = await getExtrinsic(params);
-  const feeWrappedExtrinsic = await wrapWithTxFeeSwap(feeCurrency, baseExtrinsic, pools);
-
-  return submitTransaction(window.bridge.api, params.accountAddress, feeWrappedExtrinsic, expectedStatus, params.events);
-};
+  const finalExtrinsic = wrapWithTxFeeSwap(feeAmount, baseExtrinsic, pools);
 
-type UseTransactionOptions = Omit<
-  UseMutationOptions<TransactionResult, Error, TransactionActions, unknown>,
-  'mutationFn'
-> & {
-  customStatus?: ExtrinsicStatus['type'];
-  onSigning?: (variables: TransactionActions) => void;
-  showSuccessModal?: boolean;
+  return submitTransaction(window.bridge.api, params.accountAddress, finalExtrinsic, expectedStatus, params.events);
 };
 
 // The three declared functions are use to infer types on diferent implementations
-function useTransaction<T extends Transaction>(
-  type: T,
-  options?: UseTransactionOptions
-): Exclude<UseTransactionResult<T>, ExecuteTypeArgs<T>>;
-function useTransaction<T extends Transaction>(
-  options?: UseTransactionOptions
-): Exclude<UseTransactionResult<T>, ExecuteArgs<T>>;
+function useTransaction<T extends Transaction>(type: T, options?: UseTransactionOptions): UseTransactionWithType<T>;
+function useTransaction<T extends Transaction>(options?: UseTransactionOptions): UseTransactionWithoutType<T>;
 function useTransaction<T extends Transaction>(
   typeOrOptions?: T | UseTransactionOptions,
   options?: UseTransactionOptions
 ): UseTransactionResult<T> {
   const { state } = useSubstrate();
-
+  const { data: pools } = useGetLiquidityPools();
   const { getCurrencyFromTicker } = useGetCurrencies(true);
+  const { getBalance } = useGetBalances();
 
   const [isSigned, setSigned] = useState(false);
-  const [feeCurrency, setFeeCurrency] = useState(GOVERNANCE_TOKEN);
-  const [feeEstimate, setFeeEstimate] = useState<MonetaryAmount<CurrencyExt>>();
-  const [isFeeEstimateLoading, setIsFeeEstimateLoading] = useState(false);
 
   const { showSuccessModal, customStatus, ...mutateOptions } =
     (typeof typeOrOptions === 'string' ? options : typeOrOptions) || {};
 
+  const mutateFee: (
+    pools: Array<LiquidityPool>
+  ) => MutationFunction<FeeEstimateResult, EstimateFeeParams> = useCallback(
+    (pools) => async ({ ticker, params }) => {
+      const currency = getCurrencyFromTicker(ticker);
+
+      const actionAmount = getActionAmount(params);
+
+      const isActionAmountFeeCurrency = actionAmount && isCurrencyEqual(actionAmount.currency, currency);
+
+      const feeBalance = getBalance(currency.ticker)?.transferable;
+
+      const availableBalance = actionAmount && isActionAmountFeeCurrency ? feeBalance?.sub(actionAmount) : feeBalance;
```

Comment:
This is already checked for in  `isActionAmountFeeCurrency`
```suggestion
      const availableBalance =  isActionAmountFeeCurrency ? feeBalance?.sub(actionAmount) : feeBalance;
```

Reply:
typescript problem

![image](https://github.com/interlay/interbtc-ui/assets/43269067/009e5f4a-5f7b-4cb6-8c62-359cf55475a7)